examples/traceresponse/src/server.rs

```diff
@@ -0,0 +1,67 @@
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Request, Response, Server};
+use opentelemetry::propagation::TextMapPropagator;
+use opentelemetry::trace::{SpanKind, TraceContextExt};
+use opentelemetry::Context;
+use opentelemetry::{
+    global,
+    sdk::export::trace::stdout,
+    sdk::{
+        propagation::TraceContextPropagator,
+        trace::{self, Sampler},
+    },
+    trace::Tracer,
+};
+use opentelemetry_contrib::trace::propagator::trace_context_response::TraceContextResponsePropagator;
+use opentelemetry_http::{HeaderExtractor, HeaderInjector};
+use std::{convert::Infallible, net::SocketAddr};
+
+async fn handle(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+    let parent_cx = global::get_text_map_propagator(|propagator| {
+        propagator.extract(&HeaderExtractor(req.headers()))
+    });
+    let _cx_guard = parent_cx.attach();
+
+    let tracer = global::tracer("example/client");
+    let span = tracer
+        .span_builder("say hello")
+        .with_kind(SpanKind::Server)
+        .start(&tracer);
+
+    let cx = Context::current_with_span(span);
+
+    cx.span().add_event("handling this...", Vec::new());
```

Comment:
```suggestion
    span.add_event("handling this...", Vec::new());
```

Reply:
This change would also require importing `opentelemetry::trace::Span` which would make `span` a `BoxedSpan` as opposed to a `SpanRef`, which also means making `span` mutable because `BoxedSpan::add_event` takes a mutable reference, and moving the definition of `cx` further down because `Context::current_with_span` moves the span, and `BoxedSpan` doesn't implement `Clone`.

I don't mind doing this change, but is it really necessary, when the purpose here is to illustrate how `traceresponse` works?

Reply:
No need of changing. (I am Rust newbie, so my comments could be wrong!)